README.md

```diff
@@ -36,7 +32,19 @@ for the `providers` that have the role `producer` or `processor` assigned.
 | processing:facility     | string              | The name of the facility that produced the data. For example, `Copernicus S1 Core Ground Segment - DPA` for product of Sentinel-1 satellites. |
 | processing:software     | Map<string, string> | A dictionary with name/version for key/value describing one or more softwares that produced the data. For example, `"Sentinel-1 IPF":"002.71"` for the software that produces Sentinel-1 satellites data. |
 
-*At least one of the fields must be specified.*
+These fields can be used in a variety of places:
+
+For Items:
+- The fields are placed in the properties.
+- Additionally, STAC allows all fields to be used in the Asset Object.
+
+For Collections:
+- The fields are usually placed in the [Provider Objects](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#provider-object)
+for the `providers` that have the role `producer` or `processor` assigned. They don't need to be provided for all providers of the respective role.
+- The fields can also be used in summaries, assets or Item asset definitions.
+
+If the extension is given in the `stac_extensions` list, at least one of the fields must be specified in any of the given places listed above.
+Please note that the JSON Schema might not be able to validate this requirement in all cases due to limitations in JSON Schema.
```

Comment:
Could we make sure which exact places this requirement is validated in, or at least which places it *should* be validated in? Then we could add tests to make sure it works that way. Otherwise users of the extension have to be JSON Schema experts to work out whether their use case is even validated.

Reply:
Sure. Please provide a proposal though.

Reply:
Working on some tests which illustrate the existing (buggy) behaviour.

Reply:
See #16.

Reply:
I think I've fixed the main issues in #9.

Reply:
I assumed you'd want some text to be added to this PR which makes requirements clear by just reading the text. Otherwise, we could just merge this now.

Reply:
Theistext doesn't tell the user exactly where these properties can be used such that they will be validated, but the tests in #9 do. Maybe we could use the tests as a template for updating the documentation, so that it's precise and complete?

Reply:
And I assumed you'd provide the text as you had worked on #9 ;-)

Reply:
Can do.